From 4c794046459e776f4139655405c2925278a23e4e Mon Sep 17 00:00:00 2001
From: Claudiu Manoil <claudiu.manoil@nxp.com>
Date: Fri, 12 Apr 2019 13:29:19 +0300
Subject: [PATCH 696/706] mscc: ocelot: Don't interfere with phylib's state
 machine

Removed emulation reminiscent hacks and fixed phylib integration.

Signed-off-by: Claudiu Manoil <claudiu.manoil@nxp.com>
(cherry picked from commit 4dc57012b003ec1a177d5313eab5aedc7e8e5c58)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/mscc/ocelot.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/mscc/ocelot.c b/drivers/net/ethernet/mscc/ocelot.c
index 34f66d487bec..db675fb4944e 100644
--- a/drivers/net/ethernet/mscc/ocelot.c
+++ b/drivers/net/ethernet/mscc/ocelot.c
@@ -490,8 +490,6 @@ static int ocelot_port_open(struct net_device *dev)
 		netdev_err(dev, "Could not attach to PHY\n");
 		return err;
 	}
-	netif_carrier_on(dev);
-	ocelot->port_adjust_link(dev);
 
 	dev->phydev = port->phy;
 
@@ -504,7 +502,7 @@ static int ocelot_port_stop(struct net_device *dev)
 {
 	struct ocelot_port *port = netdev_priv(dev);
 
-	netif_carrier_off(dev);
+	phy_stop(port->phy);
 	phy_disconnect(port->phy);
 
 	dev->phydev = NULL;
-- 
2.17.1

